libbeat/docs/outputconfig.asciidoc

```diff
@@ -537,6 +572,17 @@ output:
 
 If `pretty` is set to true, events written to stdout will be nicely formatted. The default is false.
 
+===== bulk_max_size
+
+The maximum number of events to bulk into batches. If the Beat sends single events, the events
+are collected into batches. If the Beat publishes a large batch of events (larger than the value
+specified by `bulk_max_size`), the batch is split.
+
+Specifying a larger batch size can improve performance by lowering the overhead of sending events. 
+However big batch sizes can also increase processing times
+
+The default is 200.
```

Comment:
The difference is in the support for bulk-events.

Logstash and elasticsearch protocol (and output plugins) do support bulk-events. That's why we should assume some performance benefit from these plugins. At the cost of possible timeouts due to requests taking too long.

The other output plugins do not support bulk-events (besides events buffered in libbeat) and will process all events one by one, besides publisher pipeline buffering events (which adds some latency). That is the section about 'improve performance' might be somewhere between correct and wrong (would rather remove it).